### DISCUSSION TITLE
How to discern which sample a cell came from?

### DISCUSSION THREAD

**Discussion by lawsonvt** [COMBINE-lab/alevin-fry#69] [Q&A] [answered]
Hello,

Using the `usefulaf` tools, I was able to align a set of 7 samples (each a set of paired end reads), same tissue, different locations. I did this all in one command, leaving me with a combined quant matrix of all the cells across all samples. As I understand it, based on discussions about alevin / salmon. it is preferable to align all reads in one go like this for better accuracy.

However, how do I determine which cell barcodes came from which sample? I do not see any form of generated metadata that lists this. Is there a file I overlooked? A flag I need to set?

Thank you for your help.

~ Mark

**Comment by rob-p** [comment, marked as answer]
Hi @lawsonvt,

  Thanks for raising this in the Q&A!  I'd like to understand the specific experimental setup a bit more.  Are all of these originally from the same experimental sample, or are they different?  When a given `sample` has multiple lanes of sequencing data, these should be combined.  However, when an experiment has independent samples, these should be run separately.  This is because samples that are independently prepped and run will have overlapping sets of barcodes.  For example, the 10x chromium protocol uses the same set of potential barcodes for each sample — thus, if you pool together the reads from all samples, you will lose information about the same of origin of each read (i.e. the cell/UMI) because reads having the same cell barcode will be aggregated together during the `collate` step of `alevin-fry`.

  So, if these are 7 independent samples, then you should run each separately.  This will produce 7 distinct output matrices; one per sample.  If you desire, you can then "integrate" these data, accounting for things like batch effects etc.  Let me know if this answers your question.

Best,
Rob

**Reply by lawsonvt** [reply]
These are 7 independent samples, so you answered my question. Thank you so much for clarifying!
